File: php/features.php
<?php
declare(strict_types=1);

// Ziptastic SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class ZiptasticFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new ZiptasticBaseFeature();
            case "test":
                return new ZiptasticTestFeature();
            default:
                return new ZiptasticBaseFeature();
        }
    }
}

File: py/features.py
# Ziptastic SDK feature factory

from feature.base_feature import ZiptasticBaseFeature
from feature.test_feature import ZiptasticTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ZiptasticBaseFeature(),
        "test": lambda: ZiptasticTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
